src/components/single-create.ts

```diff
@@ -0,0 +1,138 @@
+import Feature from 'ol/Feature';
+import Point from 'ol/geom/Point';
+import { Vector } from 'ol/source';
+import { useStore } from '../composable/store';
+import { Vector as VectorLayer } from 'ol/layer';
+import CreateStyle from './styles/create-style';
+import { GeocityEvent } from '../utils/geocity-event';
+import SelectCreateInformationBoxController from './select-create-information-box';
+import { Control } from 'ol/control';
+import { Map } from 'ol';
+
+export default class SingleCreate {
+  currentFeature: Feature | undefined;
+
+  constructor(mapElement: HTMLDivElement) {
+    const map = useStore().getMap();
+    const vectorSource = new Vector();   
+
+    this.setupMapForCreation(map, vectorSource);
+
+    window.addEventListener('authorize-created', () => {
+      this.setupCreateElement(map, vectorSource)
+    })
+
+    window.addEventListener('remove-created-icon', () => {
+      this.setupDeleteElement(map, vectorSource)
+    })
+
+    window.addEventListener('recenter-selected-element', () => {
+      map.getView().setCenter(useStore().getSelectedFeature()?.get('geometry').getCoordinates())
+    }) 
+
+    this.longClickEvent(mapElement, map);
```

Comment:
`addLongClickEvent` or `setupLongClick` would be better probably since it adds the event listener

Reply:
done